lib/src/conflicts.rs

```diff
@@ -1016,3 +1019,176 @@ pub async fn update_from_content(
     };
     Ok(new_file_ids)
 }
+
+/// Diff hunk that may be unresolved conflicts.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ConflictDiffHunk<'input> {
+    pub kind: DiffHunkKind,
+    pub lefts: Merge<&'input BStr>,
+    pub rights: Merge<&'input BStr>,
+}
+
+/// Iterator adaptor that tries to resolve trivial conflicts for each hunk.
+pub fn conflict_diff_hunks<'input, I>(
+    diff_hunks: I,
+    num_lefts: usize,
+) -> impl Iterator<Item = ConflictDiffHunk<'input>>
+where
+    I: IntoIterator,
+    I::Item: Borrow<DiffHunk<'input>>,
+{
+    // This should align with files::merge()
+    fn try_resolve<'input>(contents: &[&'input BStr]) -> Merge<&'input BStr> {
+        if let Some(&content) = merge::trivial_merge(contents) {
+            Merge::resolved(content)
+        } else {
+            Merge::from_vec(contents)
+        }
+    }
```

Comment:
I'll probably change this iterator to not attempt to resolve conflicts.

I think it's more correct to resolve conflicts per side by using `file::merge()` (or similar function of `Merge<&BStr> -> Merge<BString>` type.) This should be done by caller. It will also help deduplicate word-level merging impls.

```rust
let lefts: Merge<_> = files::premerge(lefts);
let rights: Merge<_> = files::premerge(rights);
for hunk in conflict_diff_hunks(&Diff::by_line(lefts.iter().chain(rights.iter()), lefts.len()) {
    ...
```

Reply:
Changed to merge contents by caller.

I also moved `conflict_diff_hunks()` to files module. It no longer attempts to resolve merge conflicts.